Automattic/woocommerce-payments
# Add test mode notice in page order detail.

Fixes #334 

#### Changes proposed in this Pull Request

- Use the same foundation of https://github.com/Automattic/woocommerce-payments/pull/6373 to add a new notice for test mode in the order details. Changes for this in the JS change files, and this PHP code https://github.com/Automattic/woocommerce-payments/blob/25d96a577525c54ebbfffc64ea0e900342836365/includes/admin/class-wc-payments-admin.php#L737

- I also use this opportunity to introduce `Order_Mode` constant (enum) class, and `WC_Payments_Order_Service::WCPAY_MODE_KEY`. Changes for this is for the rest of PHP files. 









#### Testing instructions





* Check for orders in the test and dev mode, confirm that you're seeing the new notice. 
* Check for orders with dispute in the test and dev mode, see both notices: test mode and dispute. 

#### Screenshots

- Only testmode notice: 

![only-testmode](https://github.com/Automattic/woocommerce-payments/assets/10045087/7404258a-4257-4c07-a1ba-7002800bf56f)

- Both testmode and dispute notices: 

![testmode-and-dispute](https://github.com/Automattic/woocommerce-payments/assets/10045087/04ecc01a-23c0-4b8e-b084-5895fb4ceaa3)

-------------------


**Post merge**
